cdap-ui/app/cdap/components/CodeEditor/JSONEditor.tsx

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import CodeEditor from 'components/CodeEditor';
+
+interface IJSONEditorProps {
+  value: string;
+  onChange: (value: string) => void;
+}
+function JSONEditor({ value, onChange, ...restProps }: IJSONEditorProps) {
+  return (
+    <CodeEditor
+      {...restProps}
+      value={value}
+      onChange={onChange}
+      activeLineMarker={false}
+      showPrettyPrintButton={true}
+      prettyPrintFunction={(unformattedValue: string) => {
+        let v = unformattedValue;
+        try {
+          v = JSON.parse(v);
+        } catch (e) {
+          return unformattedValue;
+        }
+        return JSON.stringify(v, null, 2);
```

Comment:
if `v` is a non-json string, this will cause double quotation

Reply:
I am not sure if I follow. The `JSON.parse` and the `return` inside the `catch` block are in place to make sure `v` is `JSON`. Can you give an example of when doing JSON.stringify will cause an issue?

Reply:
nevermind, please disregard. Completely missed the return statement on the catch block.